docs/enhancements/add-feature-support-by-generic-attributes.md

```diff
@@ -0,0 +1,144 @@
+---
+title: Add feature-support-level by a generic attributes
+authors:
+  - "@eliorerz"
+creation-date: 2023-02-23
+last-updated: 2023-02-23
+---
+
+# Add feature-support-level by a generic attributes
+
+## Summary
+
+In the current implementation of Assisted Installer, there is an API that allows the user to get a list of objects
+containing a list of features attached to an OpenShift version and its support-level.
+Whenever a user wants to know if a feature is supported in an OpenShift version, the user needs to call 
+the feature-support-list API, get the entire list and filter the results to find the specific feature for that version.
+As part supporting multiple CPU architectures on Assisted Installer, we want the ability to add another dimension to the 
+returned object - CPU architecture, so we can determine if a feature is supported or not by two parameters instead of one.
+
+
+## Motivation
+Adding another attribute to the feature-support-level API is indeed required but in its current implementation there are 
+few flaws (from my perspective):
+1. The current implementation forces us to duplicate the entire features list for each cpu architecture (API change).\
+For example if for `OpenshiftVersion = 4.12` we currently hold 10 features so adding just ARM64 as a CPU architecture to
+that equation will require us to duplicate the 10 features (for that architecture) and set the support-level for both 
+x86 and arm64.
+2. The list is statically defined in assisted-service code so on new OCP version we will have to duplicate the entire 
+features from the previous version even if it supported or not supported for few past versions.
+3. It is hard to expand, beside of adding CPU architecture to that equation if in the future we will want to filter by a
+third attribute we will have to duplicate everything again (API change).
+4. The user is have to get the entire feature-support list even if he needs to know just the support of a single feature.
+
+These four reasons led me to the realization that we must create a more flexible and easier-to-maintain API. 
+
+### Goals
+
+- Add another dimension (of CPU architecture) to the feature-supported endpoint  
+- Improve the maintainability of this code and less likely to require updates
+
+### Non-Goals
+
+- Change the API to allow the user to get the support level of a single feature without retrieving the entire list.
+- Design an easier way to add a new dimension to that API.
+- Get per cluster feature-support level
+
+## Proposal
```

Comment:
Here is an alternate proposal.  We can keep the current API but add a cpu_architecture query param that defaults to x86_64.  Then for x86_64, there is no change in the arguments or returned JSON.  We allow users to pass a different CPU architecture, returning a JSON with the same structure but different values.  In the future, we can add more query params without breaking the API.

Reply:
If the feature support levels can change based on CPU architecture, what would happen when there will be Day1 clusters with multiple CPU architectures? Or this could happen ATM with Day2 clusters which may have mixed CPU architectures.

It would be hard to express if a feature is supported or not if the values differ for the CPU architectures that the cluster might have. And it would imply that the UI / consuming application need to perform this logic which may not be simple.

For that reason, I find Elior's suggestion about being able to request per-cluster feature-support-levels.

Reply:
The enhancement doesn't describe how feature support for heterogeneous clusters - @eliorerz please address this.

Reply:
ok, I didn't know this was the arrangement 
Do UI follows this path? @jkilzi

Reply:
I also didn't know that one  :) 
Anyway, For future release the user will always get automatically the return value that defined in `GetSupportLevel` function